docs/guide/getting-started/recovery-example.md

```diff
@@ -76,14 +59,24 @@ total_sum_stream = op.stateful_map("running_count", keyed_stream, update_sum)
 op.output("out", total_sum_stream, StdOutSink())
 ```
 
-To enable recovery when you execute a dataflow, pass the `-r` flag to
-{py:obj}`bytewax.run` and specify the recovery directory. We will also
-need to set two values for recovery, the `snapshot_interval` via the
-`-s` flag and the `backup_interval` via the `-b` flag.
-
-The `snapshot_interval` specifies the amount of time in seconds to wait
-before creating a snapshot. The `backup_interval` specifies the amount
-of time in seconds to keep older state snapshots around.
+To enable recovery when you execute a dataflow, you need to pass some
+parameters to {py:obj}`bytewax.run`.
+- The `-d/--local-state-dir` flag to specify the directory where bytewax
+  will write the database file for that process.
+- The `-b/--backup` (optional) to indicate which `Backup` strategy to use for durable backup.
+  This defaults to a `FileSystemBackup` instance saved in a subdirectory named `backup`
+  in the same folder where the script is executed. You need to create that directory
+  or pass a different one here. This flag works the same way as the dataflow import string,
+  so you can use `--backup "bytewax.backup:file_system_backup('/tmp/bytewax/backup')"`.
+- The `--snapshot-mode` (optional) to set when bytewax should take the snapshots. There are
+  two modes: 'immediate' or 'batch'. In 'immediate' mode snapshots are taken as soon as
+  the state changes, in 'batch' mode they are taken in batches at the end of the
+  `snapshot_interval`. It's a tradeoff between latency and overall throughput. Taking
+  snapshots in batch is usually more efficient overall, but can add a delay between
+  snapshot intervals. Defaults to `immediate`.
+- The `-s/--snapshot-interval` indicates when the recovery system should take snapshots if
+  the `snapshot-mode` is `batch`, and when the `backup` class should be called to save the state
+  into durable backup.
```

Comment:
The way its implemented, it actually gives continuous resume points, but that's because the compactor is not actually compacting (see comment below), I'll fix this docstring when fixing the rest

Reply:
Sorry, I got this wrong. The way it works it actually creates continuous resume points, but that's not due to the compactor, I just got the immediate_snapshot behavior wrong. The mental model is clear, but I somehow lost focus on this point while doing the rest of the work. The actual bug seems to be a combination of an off by one error in the frontier written and the wrong condition on the query used to get the snapshots back from the local state store on a fast resume. I'm fixing this